### ISSUE TITLE
Error "Must declare a coordinate system services object"

### ISSUE THREAD

**Issue by jtoomey** [SharpMap/SharpMap#225]
Performing these 2 steps
`var map = new SharpMap.Map(new System.Drawing.Size(1024, 786));`
`var provider = new SharpMap.Data.Providers.ShapeFile("path_to_shapefile", true);`
elicits the error "Must declare a coordinate system services object" on the 2nd line. I tried with a few different shapefiles so it doesn't appear related to the file itself. Is there a step that must be performed first? 
This same code works with SharpMap ver 1.1, and also an old version of SharpMap 0.9 that we have from a project done years back, so something seems to have changed with 1.2.

**Comment by PABLOCORRO** [comment]
Hello
try this code:


ProjNet.CoordinateSystems.CoordinateSystemFactory csf = new ProjNet.CoordinateSystems.CoordinateSystemFactory();
            ProjNet.CoordinateSystems.Transformations.CoordinateTransformationFactory ctf = new ProjNet.CoordinateSystems.Transformations.CoordinateTransformationFactory();
            Session.Instance.SetCoordinateSystemServices(new SharpMap.CoordinateSystems.CoordinateSystemServices(csf, ctf, SharpMap.Converters.WellKnownText.SpatialReference.GetAllReferenceSystems()));

**Comment by jtoomey** [comment]
Thank you! Yes this worked. 
One question:
I'm using Nuget to pull in SharpMap, so I also had to bring in GeoAPI and ProjNET. I can use the current version 1.7.5 of GeoAPI, but ProjNET's current version 2.0 made changes to CoordinateSystemFactory and CoordinateTransformationFactory, so I had to manually download the older version 1.4 of ProjNET to get it to work.
Does this sound right? I should stick with 1.4 of ProjNET for now?
